base/range.jl

```diff
@@ -963,14 +963,24 @@ end
 # This is separate to make it useful even when running with --check-bounds=yes
 function unsafe_getindex(r::StepRangeLen{T}, i::Integer) where T
     i isa Bool && throw(ArgumentError("invalid index: $i of type Bool"))
-    u = i - r.offset
```

Comment:
I wonder if a better fix would be
```julia
    u = oftype(r.offset, i) - r.offset
```
since if the number wasn't representable as the type of the offset, then it was out of bounds anyways. and this cast should typically be free.